src/app/workflow/tool-tab/tool-tab.component.ts

```diff
@@ -35,12 +34,13 @@ export class ToolTabComponent extends EntryTab {
   ToolDescriptor = ToolDescriptor;
   @Input() set selectedVersion(value: WorkflowVersion) {
     if (value != null) {
-        this.workflow = this.workflowQuery.getActive();
-        if (this.workflow) {
-          this.updateTableToolContent(this.workflow.id, value.id);
-        } else {
-          console.error('Should not be able to select version without a workflow');
-        }
+      this.workflow = this.workflowQuery.getActive();
+      // Also check that the workflow version belongs to the workflow
+      if (this.workflow && this.workflow.workflowVersions.some(version => version.id === value.id)) {
```

Comment:
Is `workflowVersions` guaranteed not to be null?

Reply:
value is checked above